example/otel-collector/main.go

```diff
@@ -58,6 +58,8 @@ func initProvider() (func(context.Context) error, error) {
 	// `localhost:30080` endpoint. Otherwise, replace `localhost` with the
 	// endpoint of your cluster. If you run the app inside k8s, then you can
 	// probably connect directly to the service through dns
+	ctx, cancel := context.WithTimeout(ctx, time.Second)
+	defer cancel()
```

Comment:
@dmathieu I understand that PR is already merged, but won't it cause any issues that context is canceled when `initProvider` function returns?

Reply:
Hi,
That is possible. Unfortunately, I won't be able to look into this in the next couple weeks. If you can reproduce the problem when running the example, feel free to open a fix.